.github/actions/setup-build/action.yml

```diff
@@ -0,0 +1,226 @@
+          "" { $TargetArch = $HostArch }
+          default {
+            Write-Output "::error::Unsupported target architecture: `"${{ inputs.target-arch }}`""
+            exit 1
+          }
+        }
+
+        Write-Output "Host OS: $HostOs"
+        Write-Output "Host architecture: $HostArch"
+        Write-Output "Target architecture: $TargetArch"
+
+        $VerifiedInput = @"
+        host-os=$HostOs
+        host-arch=$HostArch
+        target-arch=$TargetArch
+        "@
+        Write-Output $VerifiedInput | Out-File -FilePath $env:GITHUB_OUTPUT -Encoding utf8 -Append
+
+    - name: Install Windows SDK version ${{ inputs.windows-sdk-version }}
+      if: steps.verify-input.outputs.host-os == 'windows' && inputs.windows-sdk-version != ''
+      shell: pwsh
+      run: |
+        $WinSdkVersionString = "${{ inputs.windows-sdk-version }}"
+        $WinSdkVersion = [System.Version]::Parse($WinSdkVersionString)
+        $WinSdkVersionBuild = $WinSdkVersion.Build
+
+        $Win10SdkRoot = Get-ItemPropertyValue `
+          -Path "Registry::HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows Kits\Installed Roots" `
+          -Name "KitsRoot10"
+        $Win10SdkLib = Join-Path $Win10SdkRoot "Lib"
+        $Win10SdkInclude = Join-Path $Win10SdkRoot "Include"
+        $Win10SdkIncludeVersion = Join-Path $Win10SdkInclude $WinSdkVersionString
+
+        if (Test-Path -Path $Win10SdkIncludeVersion -PathType Container) {
+          Write-Output "Windows SDK ${WinSdkVersionString} already installed."
+        } else {
+          # Install the missing SDK.
+          Write-Output "Installing Windows SDK ${WinSdkVersionString}..."
+
+          $InstallerLocation = Join-Path "${env:ProgramFiles(x86)}" "Microsoft Visual Studio" "Installer"
+          $VsWhere = Join-Path "${InstallerLocation}" "vswhere.exe"
+          $VsInstaller = Join-Path "${InstallerLocation}" "vs_installer.exe"
+          $InstallPath = (& "$VsWhere" -latest -products * -format json | ConvertFrom-Json).installationPath
+          $process = Start-Process "$VsInstaller" `
+              -PassThru `
+              -ArgumentList "modify", `
+                  "--installPath", "`"$InstallPath`"", `
+                  "--channelId", "https://aka.ms/vs/17/release/channel", `
+                  "--quiet", "--norestart", "--nocache", `
+                  "--add", "Microsoft.VisualStudio.Component.Windows11SDK.${WinSdkVersionBuild}"
+          $process.WaitForExit()
+
+          if (Test-Path -Path $Win10SdkIncludeVersion -PathType Container) {
+            Write-Output "Windows SDK ${WinSdkVersionString} installed successfully."
+          } else {
+            Write-Output "::error::Failed to install Windows SDK ${WinSdkVersionString}."
+            Write-Output "Installer log:"
+            $log = Get-ChildItem "${env:TEMP}" -Filter "dd_installer_*.log" | Sort-Object LastWriteTime -Descending | Select-Object -First 1
+            Get-Content $log.FullName
```

Comment:
I think that it is better to attach the log rather than print it. The GH UI doesn't deal well with large blobs of text.

Reply:
I'll do that in a follow-up PR, before I start using the action in production. It's a bit tricky because we have to export the file name before we can feed it to the upload step.